### DISCUSSION TITLE
How does react's `cache` work?

### DISCUSSION THREAD

**Discussion by darklight9811** [vercel/next.js#47965] [Help]
### Summary

I'm trying to understand the cache method described [here](https://beta.nextjs.org/docs/data-fetching/caching#per-request-caching), and it looks like is rerunning the method even if the arguments are the same. An example:

``` typescript
import { cache } from 'react';

const cb1 = cache(async function (id: string) {
  console.log('cache ' + id);
});

function Comp() {
  cb1('a');

  return null;
}

export default function Home() {
  cb1('a');

  return <Comp />;
}
```

Console log will be called twice, meaning that even if the arguments are the same, it still reruns the method.

**Comment by rehanzfr** [comment]
Now, after updating to the Next 13, I am having this error during production. 

Type error: Module '"react"' has no exported member 'cache'.

Is there any help available regarding this matter?

**Reply by daniilgri** [reply]
First of all, `cache` is only for use inside Server components.
Second of all, `cache` is only available in React’s [Canary](https://react.dev/community/versioning-policy#canary-channel) and [experimental](https://react.dev/community/versioning-policy#experimental-channel) channels.
